### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by DanielFran**

### CODE REVIEW

**Comment by DanielFran** [comment]
@mshima To reuse build-matrix from generator project we need some updates on the path for the 99-build-changes.sh script?

**Comment by mshima** [comment]
I will take a look, just found a but at build-changes script.

File: .github/workflows/angular.yml [hunk 1]
```diff
@@ -32,9 +32,37 @@ on:
...
+        with:
+          repository: 'jhipster/generator-jhipster'
+          path: generator-jhipster
+          fetch-depth: 2
+          ref: main
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
+        with:
+          workflow-samples-file: angular
```
Comment (mshima) [inline .github/workflows/angular.yml:61]: Changes detection won't work here.
```suggestion
          generator-jhipster-directory: ${{ github.workspace }}/generator-jhipster
          skip-changes-detection: true
          workflow-samples-file: angular
```

File: .github/workflows/react.yml [hunk 2]
```diff
@@ -32,9 +32,37 @@ on:
...
+        with:
+          repository: 'jhipster/generator-jhipster'
+          path: generator-jhipster
+          fetch-depth: 2
+          ref: main
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
+        with:
+          workflow-samples-file: workflow-react
```
Comment (mshima) [inline .github/workflows/react.yml:61]:
```suggestion
          generator-jhipster-directory: ${{ github.workspace }}/generator-jhipster
          skip-changes-detection: true
          workflow-samples-file: react
```

File: .github/workflows/vue.yml [hunk 3]
```diff
@@ -32,9 +32,37 @@ on:
...
+        with:
+          repository: 'jhipster/generator-jhipster'
+          path: generator-jhipster
+          fetch-depth: 2
+          ref: main
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
+        with:
+          workflow-samples-file: workflow-vue
```
Comment (mshima) [inline .github/workflows/vue.yml:61]:
```suggestion
          generator-jhipster-directory: ${{ github.workspace }}/generator-jhipster
          skip-changes-detection: true
          workflow-samples-file: vue
```

File: .github/workflows/vue.yml [hunk 4]
```diff
@@ -32,9 +32,37 @@ on:
...
+      server: ${{ steps.build.outputs.server }}
+      any: ${{ steps.build.outputs.any }}
+      workflow-angular: ${{ steps.build.outputs.workflow-angular }}
+      workflow-react: ${{ steps.build.outputs.workflow-react }}
+      workflow-vue: ${{ steps.build.outputs.workflow-vue }}
+    steps:
+      - name: 'SETUP: Checkout jhipster-bom'
+        uses: actions/checkout@v3
+        with:
+          path: jhipster-bom
```
Comment (mshima) [inline .github/workflows/vue.yml:49]:
```suggestion
```

File: .github/workflows/angular.yml [hunk 5]
```diff
@@ -32,9 +32,37 @@ on:
...
+      server: ${{ steps.build.outputs.server }}
+      any: ${{ steps.build.outputs.any }}
+      workflow-angular: ${{ steps.build.outputs.workflow-angular }}
+      workflow-react: ${{ steps.build.outputs.workflow-react }}
+      workflow-vue: ${{ steps.build.outputs.workflow-vue }}
+    steps:
+      - name: 'SETUP: Checkout jhipster-bom'
+        uses: actions/checkout@v3
+        with:
+          path: jhipster-bom
```
Comment (mshima) [inline .github/workflows/angular.yml:49]: Not needed
```suggestion
```

File: .github/workflows/react.yml [hunk 6]
```diff
@@ -32,9 +32,37 @@ on:
...
+      server: ${{ steps.build.outputs.server }}
+      any: ${{ steps.build.outputs.any }}
+      workflow-angular: ${{ steps.build.outputs.workflow-angular }}
+      workflow-react: ${{ steps.build.outputs.workflow-react }}
+      workflow-vue: ${{ steps.build.outputs.workflow-vue }}
+    steps:
+      - name: 'SETUP: Checkout jhipster-bom'
+        uses: actions/checkout@v3
+        with:
+          path: jhipster-bom
```
Comment (mshima) [inline .github/workflows/react.yml:49]:
```suggestion
```

File: .github/workflows/angular.yml [hunk 7]
```diff
@@ -32,9 +32,37 @@ on:
...
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+          path: generator-jhipster
+          fetch-depth: 2
+          ref: main
```
Comment (mshima) [inline .github/workflows/angular.yml:56]:
```suggestion
```

File: .github/workflows/vue.yml [hunk 8]
```diff
@@ -32,9 +32,37 @@ on:
...
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+          path: generator-jhipster
+          fetch-depth: 2
+          ref: main
```
Comment (mshima) [inline .github/workflows/vue.yml:56]:
```suggestion
```

File: .github/workflows/react.yml [hunk 9]
```diff
@@ -32,9 +32,37 @@ on:
...
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+          path: generator-jhipster
+          fetch-depth: 2
+          ref: main
```
Comment (mshima) [inline .github/workflows/react.yml:56]:
```suggestion
```

File: .github/workflows/vue.yml [hunk 10]
```diff
@@ -32,9 +32,31 @@ on:
...
+      workflow-react: ${{ steps.build.outputs.workflow-react }}
+      workflow-vue: ${{ steps.build.outputs.workflow-vue }}
+    steps:
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
```
Comment (mshima) [inline .github/workflows/vue.yml:52]:
```suggestion
        uses: ./.github/actions/build-matrix
```

File: .github/workflows/react.yml [hunk 11]
```diff
@@ -32,9 +32,31 @@ on:
...
+      workflow-react: ${{ steps.build.outputs.workflow-react }}
+      workflow-vue: ${{ steps.build.outputs.workflow-vue }}
+    steps:
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
```
Comment (mshima) [inline .github/workflows/react.yml:52]:
```suggestion
        uses: ./.github/actions/build-matrix
```

File: .github/workflows/angular.yml [hunk 12]
```diff
@@ -32,9 +32,32 @@ on:
...
+    steps:
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
+        with:
+          skip-changes-detection: true
```
Comment (mshima) [inline .github/workflows/angular.yml:54]:
```suggestion
```

File: .github/workflows/angular.yml [hunk 13]
```diff
@@ -32,9 +32,32 @@ on:
...
+      workflow-react: ${{ steps.build.outputs.workflow-react }}
+      workflow-vue: ${{ steps.build.outputs.workflow-vue }}
+    steps:
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v3
+        with:
+          repository: 'jhipster/generator-jhipster'
+      - name: 'Build matrix'
+        id: build
+        uses: ./generator-jhipster/.github/actions/build-matrix
```
Comment (mshima) [inline .github/workflows/angular.yml:52]:
```suggestion
        uses: ./.github/actions/build-matrix
```

File: .github/workflows/react.yml [hunk 14]
```diff
@@ -162,11 +169,22 @@ jobs:
         uses: actions/upload-artifact@v2
```
Comment (mshima) [inline .github/workflows/react.yml:169]: We will need to replace local scripts with npm scripts too.

**Pull request merged**